Source/Core/Common/Logging/LogManager.cpp

```diff
@@ -228,14 +231,21 @@ void LogManager::LogWithFullPath(LogLevel level, LogType type, const char* file,
   }
 }
 
-LogLevel LogManager::GetLogLevel() const
+LogLevel LogManager::GetEffectiveLogLevel() const
+{
+  return m_effective_level;
+}
+
+void LogManager::SetConfigLogLevel(const LogLevel level)
 {
-  return m_level;
+  Config::SetBaseOrCurrent(LOGGER_VERBOSITY, level);
+  SetEffectiveLogLevel();
 }
 
-void LogManager::SetLogLevel(LogLevel level)
+void LogManager::SetEffectiveLogLevel()
 {
-  m_level = std::clamp(level, LogLevel::LNOTICE, MAX_LOGLEVEL);
+  const LogLevel config_level = Config::Get(LOGGER_VERBOSITY);
+  m_effective_level = std::clamp(config_level, LogLevel::LNOTICE, MAX_EFFECTIVE_LOGLEVEL);
```

Comment:
Config changed callbacks can be called from any thread, so to be technically correct here, m_effective_level should be atomic. (Using std::memory_order_relaxed for performance should be fine.)